plugins/auth-backend/src/identity/UserInfoDatabaseHandler.ts

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { BackstageUserInfo } from '@backstage/backend-plugin-api';
+import { Knex } from 'knex';
+
+const TABLE = 'user_info';
+
+type Row = {
+  user_entity_ref: string;
+  user_info: string;
+};
+
+// TODO: How do we prune stale users?
```

Comment:
These should probably accept (and store) effectively the `exp` of the token so that they can be cleaned up automatically (after a grace period) after the token itself becomes useless. Remember, you have to present an actual valid non-expired token to be able to read out of this anyway.